optimum/habana/transformers/generation/utils.py

```diff
@@ -743,6 +750,7 @@ def generate(
         if generation_config.static_shapes:
             # Pad inputs to have static shapes during generation, this gives better performance than dynamic shapes on HPUs
             # In encoder_decoder models, Inputs are already padded
+            assert generation_config.ignore_eos, "For gaudi we pad input_ids with EOS, so this option isnt available for now"
```

Comment:
Not sure what you mean here. That `ignore_eos == False` is not compatible with static shapes?

Reply:
In some cases pad token and eos are same (like in gpt2). `ignore_eos==False` means a stopping criteria is to look for sentences generation eos token and stop. However if we pad with eos tokens, we will make a false stop. I put this assert under another check now `if generation_config.pad_token_id == generation_config.eos_token_id:`. For some models like llama where eos token != pad token, we can have ignore_eos=False